File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> a ,b;
        if(s.size()!=t.size()) return false;
        for(char c:s){
            a[c]++;
        }
        // for(char c:t){
        //     b[c]++;
        // }
        int count = s.length();
        for(char c:t){
            if(a.find(c)==0){
                return false;
            }
            a[c]--;
            count--;
        }
        for(auto it:a){
            if(it.second!=0){
                return false;
            }
        }
        if(count!=0){
            return false;
        }
        // return a==b;
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        # return Counter(s)==Counter(t)
        if len(s) != len(t):
            return False
        
        my_dict = defaultdict(int)
        for char in s:
            my_dict[char] += 1
        for char in t:
            if my_dict[char]==0:
                return False
            my_dict[char] -= 1
        
        return True